app/components/app_patient_programmes_table_component.rb

```diff
@@ -0,0 +1,182 @@
+# frozen_string_literal: true
+
+class AppPatientProgrammesTableComponent < ViewComponent::Base
+  def initialize(patient, programmes:)
+    super
+
+    @patient = patient
+    @programmes = programmes
+  end
+
+  def call
+    govuk_table(
+      caption: CAPTION,
+      head: HEADERS,
+      rows:,
+      first_cell_is_header: true
+    )
+  end
+
+  private
+
+  attr_reader :patient, :programmes
+
+  CAPTION = "Vaccination programmes"
+  HEADERS = ["Programme name", "Status", "Notes"].freeze
+
+  def rows
+    programmes.flat_map { |programme| rows_for_programme(programme) }
+  end
+
+  def rows_for_programme(programme)
+    if programme.seasonal?
+      seasonal_programme_rows(programme)
+    else
+      non_seasonal_programme_rows(programme)
+    end
+  end
+
+  def seasonal_programme_rows(programme)
+    eligible_year_groups = eligible_year_groups_for(programme:)
+    rows = []
+
+    AcademicYear.all.each do |academic_year| # rubocop:disable Rails/FindEach
```

Comment:
```suggestion
    AcademicYear.all.flat_map do |academic_year| # rubocop:disable Rails/FindEach
```

That way you can return arrays and don't need to append to `rows`, and then they will be concatenated together.